src/main/java/ch/njol/skript/expressions/ExprStructureSectionInfo.java

```diff
@@ -0,0 +1,181 @@
+			"If you registered your own structure through Skript, you can use 0 as it won't have multiple pallets, " +
+			"this is also the default value and this setting is not required in the section.",
+	"- mirror the mirror setting for the structure on placement.",
+	"All setting syntaxes are optional and this doesn't even need to be a section, can be an effect. " +
+			"Default settings are rotation = none, pallet = 0, mirror = none, entities = true, integrity = 1",
+	"Note that some Minecraft structures have conditions that Mojang has defined, such as an ancient city will spawn near bedrock level."
+})
+@Examples({
+	"place structure \"Example\" at {_location}",
+	"place structure \"minecraft:ancient_city\" at {_location} without entities # Places near bedrock level",
+	"place structure \"minecraft:end_city\" at player's location:",
+		"\tset pallet to -1 # Will pick a random variation",
+	"place structure \"minecraft:end_city\" at player's location:",
+		"\tset includes entities to false",
+		"\tset integrity to 0.9",
+		"\tset pallet to 2",
+		"\tset rotation to clockwise 90",
+		"\tset mirror to left to right"
+})
+@RequiredPlugins("Minecraft 1.17.1+")
+@Since("INSERT VERSION")
+public class ExprStructureSectionInfo extends SimpleExpression<Object> {
+
+	static {
+		if (Skript.classExists("org.bukkit.structure.Structure")) {
+			Skript.registerExpression(ExprStructureSectionInfo.class, Object.class, ExpressionType.SIMPLE,
+					"includes entities", "rotation", "integrity", "pallet [index]", "mirror"
+			);
+		}
+	}
+
+	private enum Setting {
+
+		INCLUDES_ENTITIES(Boolean.class, StructurePlaceEvent::setIncludesEntities),
+		ROTATION(StructureRotation.class, StructurePlaceEvent::setRotation),
+		INTEGRITY(Float.class, (details, number) -> {
+			float integrity = ((Number) number).floatValue();
+			details.setIntegrity(Math.min(1, Math.max(0, integrity)));
+		}),
+		PALLET(Integer.class, (details, number) -> {
+			int pallet = ((Number) number).intValue();
+			int max = details.getStructure().getPaletteCount();
+			details.setPallet(Math.min(max, Math.max(-1, pallet)));
+		}),
+		MIRROR(Mirror.class, StructurePlaceEvent::setMirror);
+
+		private final BiConsumer<StructurePlaceEvent, Object> consumer;
+		private final Class<? extends Object> returnType;
+
+		@SuppressWarnings("unchecked")
+		<T> Setting(Class<T> returnType, BiConsumer<StructurePlaceEvent, T> consumer) {
+			this.returnType = returnType;
+			this.consumer = (BiConsumer<StructurePlaceEvent, Object>) consumer;
+		}
+
+		public void accept(StructurePlaceEvent event, Object object) {
+			consumer.accept(event, object);
+		}
+
+		public Class<? extends Object> getReturnType() {
+			return returnType;
+		}
+
+	}
+
+	private Setting setting;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!getParser().isCurrentSection(EffSecStructurePlace.class)) {
```

Comment:
```suggestion
		if (!getParser().isCurrentEvent(StructurePlaceEvent.class)) {
```
Checking the current section will not work because you are loading the section into a trigger (and thus it is separated from the main trigger)

Reply:
Well Skript needs a proper API for custom data in sections.

Reply:
Agreed but this syntax isn't usable right now and this fix should work alright for now